src/appshell/iappshellconfiguration.h

```diff
@@ -66,6 +66,9 @@ class IAppShellConfiguration : MODULE_EXPORT_INTERFACE
     virtual bool needShowSplashScreen() const = 0;
     virtual void setNeedShowSplashScreen(bool show) = 0;
 
+    virtual QString preferencesDialogLastOpenedPageId() const = 0;
+    virtual void setPreferencesDialogLastOpenedPageId(QString lastOpenedPageId) = 0;
```

Comment:
In this case, a "const reference" can be used in both methods, to prevent unnecessarily creating a copy of the QString:
```suggestion
    virtual const QString& preferencesDialogLastOpenedPageId() const = 0;
    virtual void setPreferencesDialogLastOpenedPageId(const QString& lastOpenedPageId) = 0;
```

Reply:
Done